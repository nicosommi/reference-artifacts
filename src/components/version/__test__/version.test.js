// @flow
import * as React from "react"

import { JestMockFn } from "jest"
import Version from "../"

import type { Props, State } from "../"
import { ShallowWrapper, shallow } from "enzyme"

describe("Version component ", () => {
  describe("onTouch", () => {
    let spy: JestMockFn
    let wrapper: ShallowWrapper<Version, Props, State>

    beforeEach(() => {
      spy = jest.fn()
      wrapper = shallow<Version, Props, State>(
        <Version version="2.354.3347-alpha4" onPress={spy} />
      )
      wrapper.find("Button").simulate("press")
    })

    it("should render the version prop", () => {
      expect(spy.mock.calls.length).toBe(1)
    })

    it("should increment the clicks", () => {
      expect(wrapper.state().clicks).toBe(1)
    })

    it("should refresh clicks after a touch", () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
