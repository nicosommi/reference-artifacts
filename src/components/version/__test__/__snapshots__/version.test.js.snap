// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Version component  onTouch should refresh clicks after a touch 1`] = `
<Fragment>
  <Button
    onPress={[Function]}
    title="Press me"
  />
  <Text>
    Version 
    2.354.3347-alpha4
  </Text>
  <Text>
    Clicks 
    1
  </Text>
</Fragment>
`;
